File: mobile/app/src/main/java/com/example/derdine/data/model/Thread.kt
package com.example.derdine.data.model

data class Thread(
    val id: String,
    val title: String,
    val content: String,
    val author: User,
    val category: Category,
    val createAt: Long,
    val viewCount: Int = 0,
    val replyCount: Int = 0,
    val likeCount: Int = 0,
    val isPinned: Boolean = false,
    val isLocked: Boolean = false
)

File: backend/src/models/Thread.ts
import mongoose from 'mongoose';

const ThreadSchema = new mongoose.Schema({
    title: { type: String, required: true },
    content: { type: String, required: true },
    author: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    category: { type: mongoose.Schema.Types.ObjectId, ref: 'Category', required: true },
    viewCount: { type: Number, default: 0 },
    replyCount: { type: Number, default: 0 },
    likeCount: { type: Number, default: 0 },
    likes: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }],
    isPinned: { type: Boolean, default: false },
    isLocked: { type: Boolean, default: false },
    createdAt: { type: Date, default: Date.now },
    updatedAt: { type: Date, default: Date.now }
});

export const Thread = mongoose.model('Thread', ThreadSchema);
